db/active_replicator.go

```diff
@@ -85,12 +85,12 @@ func (ar *ActiveReplicator) Stop() error {
 
 	var pushErr error
 	if ar.Push != nil {
-		pushErr = ar.Push.Stop()
+		pushErr = ar.Push.stop()
```

Comment:
I don't think that having two different stop methods, one private and one public, is the best way to avoid the function collision  here.  Particularly when a call to the private stop() ends up invoking the public Stop().  I'd recommend making the externally invoked push and pull implementations just Stop(), and use something more descriptive for the internally used common stop.  Given the functionality, stopAndDisconnect() seems reasonable.

Reply:
Sure Adam, added changes to make the externally invoked push and pull implementations just Stop() and internally used common stop to stopAndDisconnect().